plugins/woocommerce/includes/legacy/class-wc-legacy-api.php

```diff
@@ -90,30 +146,72 @@ public function handle_rest_api_requests() {
 			$wp->query_vars['wc-api-route'] = $_GET['wc-api-route'];
 		}
 
+		if ( empty( $wp->query_vars['wc-api-version'] ) || empty( $wp->query_vars['wc-api-route'] ) ) {
+			return;
+		}
+
 		// REST API request.
-		if ( ! empty( $wp->query_vars['wc-api-version'] ) && ! empty( $wp->query_vars['wc-api-route'] ) ) {
 
-			wc_maybe_define_constant( 'WC_API_REQUEST', true );
-			wc_maybe_define_constant( 'WC_API_REQUEST_VERSION', absint( $wp->query_vars['wc-api-version'] ) );
+		wc_maybe_define_constant( 'WC_API_REQUEST', true );
+		wc_maybe_define_constant( 'WC_API_REQUEST_VERSION', absint( $wp->query_vars['wc-api-version'] ) );
+
+		$route =  $wp->query_vars['wc-api-route'];
+		$this->maybe_log_rest_api_request($route, $_SERVER['HTTP_USER_AGENT'] ?? null);
 
-			// Legacy v1 API request.
-			if ( 1 === WC_API_REQUEST_VERSION ) {
-				$this->handle_v1_rest_api_request();
-			} elseif ( 2 === WC_API_REQUEST_VERSION ) {
-				$this->handle_v2_rest_api_request();
-			} else {
-				$this->includes();
+		// Legacy v1 API request.
+		if ( 1 === WC_API_REQUEST_VERSION ) {
+			$this->handle_v1_rest_api_request();
+		} elseif ( 2 === WC_API_REQUEST_VERSION ) {
+			$this->handle_v2_rest_api_request();
+		} else {
+			$this->includes();
 
-				$this->server = new WC_API_Server( $wp->query_vars['wc-api-route'] );
+			$this->server = new WC_API_Server( $route );
 
-				// load API resource classes.
-				$this->register_resources( $this->server );
+			// load API resource classes.
+			$this->register_resources( $this->server );
 
-				// Fire off the request.
-				$this->server->serve_request();
+			// Fire off the request.
+			$this->server->serve_request();
+		}
+	}
+
+	/**
+	 * Display an admin notice with information about the last Legacy REST API usage,
+	 * if the corresponding transient is available and unless the Legacy REST API
+	 * extension is installed or the user has dismissed the notice.
+	 */
+	private function maybe_display_legacy_wc_api_usage_notice(): void {
+		if ( is_plugin_active( 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php' ) ) {
+			if ( WC_Admin_Notices::has_notice( 'legacy_api_usages_detected' ) ) {
+				WC_Admin_Notices::remove_notice( 'legacy_api_usages_detected', true );
```

Comment:
I think we should also remove the notice if the transient doesn't exist. Let's say you actually fix your integrations and no new legacy requests are happening.
2 weeks after that WP would delete the transient itself, yet this warning will never go away unless you explicitly click "Dismiss". I believe if the transient doesn't exist, it'd be good to clear the notice too.

Reply:
> Yet this warning will never go away unless you explicitly click "Dismiss"

Yes, I implemented it that way on purpose. Removing the Legacy REST API on a system where it is in use is a big deal, and I prefer to be "too noisy" and force the user to acknowlegde the information by either dismissing the notice or installing the extension.

> Similarly, we shouldn't probably show this if the Legacy REST API is actually disabled

Same reasoning, the legacy REST API might be disabled but the notice appearing means that it has been enabled (and in use) in the past.

Reply:
I understand that but I think this can be confusing.

If nothing has happened in 2 weeks, why show the message? Yeah, it was enabled in the past but no request has come through in 2 weeks. If that happens, the notice will re-appear, so we're not hiding any information.

Also, if a merchant explicitly disables the legacy REST API, then showing the notice could give them the idea that it wasn't truly disabled and requests are still coming. That is, unless they are keeping track of the date they disabled it and compare it with the one in the notice.

I don't think this is necessarily a blocker, but I do think it's a weird UX.

Reply:
Ok, I have modified the code so that if either the transient exists, the legacy API is enabled or the separate extension is active, the notice won't appear.